### ISSUE TITLE
How to create a Mask in stage 1 to improve model when training

### ISSUE THREAD

**Issue by RyanPham19092002** [GANWANSHUI/GaussianOcc#29]
Hi. Thanks for your great work.

I'm try to create your stage 1 of your model.
I'm create a neural network to training opacity map, rotation map, scaling map based on model GPS-Gaussian. After that I give it to Gaussian Splatting to training stage1.

However, the results are bad when training like the image below:
![image](https://github.com/user-attachments/assets/a585ae31-7e64-43cf-9b70-c1adfde9875b)
The image when render in novel view:
![image](https://github.com/user-attachments/assets/946d82b1-8cf2-4ef7-af77-0ac7dddfa056)

How can I create a mask like the paper mentioned to improve the result training? Could you please give me the paper or related document use this technique ?

Thank you so much and wishing you have a great day.

**Comment by GANWANSHUI** [comment]
It was a bit long to give the details, in summary, we use this [function](https://github.com/GANWANSHUI/GaussianOcc/blob/db0b37e341bef841ea7550a6a9c0c8e5f80b98cc/networks/occupancy_decoder.py#L205) to determine whether the sampled points are in the overlap region or not. then, render it back to the image plane by summing all the points in the rays to determine whether the pixel is in an overlap region or not.

**Comment by RyanPham19092002** [comment]
Hi, thank you for your reply.

I have 4 additional questions about this function:

1. Why you use feature from encoder to determine the sampled points are in the overlap region (and then render it back to determine the pixel is in overlap region or not) instead of using directly rgb input?

2. Do you have any workaround instead of using voxel to determine the pixel in an overlap region? (I don't want to use voxel in stage 2, I just want to use the rgb image and feature to render a novel view at stage 1)

3. If there is just one way to determine the pixel in the overlap region by voxel from the feature image (like your code, which shape of the **mask_mems** is [B,S,C,Z,Y,X] , B = 1, S = 6 and **feat_mem** after reduce mask has shape [B,C,Z,Y,X], B = 1), how can I project this feature again into an image that creates a mask image like in your paper? 
![image](https://github.com/user-attachments/assets/bc1fe293-b70b-4e94-8b3d-6972587b3ed1)

4. Could you please give me any documents or paper to determine pixel in overlap region that you refer ?

Thank you for your time. Looking forward to hearing you soon
